src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -69,9 +70,24 @@ public void Dispose()
         protected virtual void Dispose(bool disposing)
         {
             // No managed resources to clean up, so disposing is ignored.
+            try
+            {
+                try
+                {
+                    Directory.Delete(TestDirectory, recursive: true);
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    DirectoryInfo di = new DirectoryInfo(TestDirectory);
+                    foreach (FileInfo file in di.GetFiles("*", SearchOption.AllDirectories))
+                    {
+                        file.IsReadOnly = false;
+                    }
 
-            try { Directory.Delete(TestDirectory, recursive: true); }
-            catch { } // avoid exceptions escaping Dispose
+                    Directory.Delete(TestDirectory, recursive: true);
+                }
+            }
+            catch {  } // avoid exceptions escaping Dispose
```

Comment:
You can do this instead, unless you want to also catch exceptions occurring in the `catch (UnauthorizedAccessException)` block.
```cs
try 
{ 
    Directory.Delete(TestDirectory, recursive: true); 
}
catch (UnauthorizedAccessException)
{
    DirectoryInfo di = new DirectoryInfo(TestDirectory);
    foreach (FileInfo file in di.GetFiles("*", SearchOption.AllDirectories))
    {
        file.IsReadOnly = false;
    }
}
catch { } // avoid exceptions escaping Dispose

```

Reply:
can you clarify? that looks the same as what I have, except that it doesn't delete the directory.

yes, I think we do want to catch exceptions from this block. we have decided that failing to cleanup should not fail the test.

Reply:
> can you clarify? that looks the same as what I have, except that it doesn't delete the directory.

I was pointing out that you could avoid nested `try`s. But since you want to try-catch the UnauthorizedAccessException case, you still need the nested try. You can ignore.